controllers/operandrequest/reconcile_operand.go

```diff
@@ -170,17 +155,31 @@ func (r *Reconciler) reconcileOperand(ctx context.Context, requestInstance *oper
 					klog.Errorf("installplan %s/%s is failed", ipNamespace, ipName)
 					requestInstance.SetMemberStatus(operand.Name, operatorv1alpha1.OperatorFailed, "", &r.Mutex)
 					continue
-				} else if ip.Status.Phase == olmv1alpha1.InstallPlanPhaseRequiresApproval {
-					klog.V(2).Infof("Installplan %s/%s is waiting for approval", ipNamespace, ipName)
-					requestInstance.SetMemberStatus(operand.Name, operatorv1alpha1.OperatorInstalling, "", &r.Mutex)
-					continue
-				} else if ip.Status.Phase != olmv1alpha1.InstallPlanPhaseComplete {
-					klog.Warningf("Installplan %s/%s is not ready", ipNamespace, ipName)
-					requestInstance.SetMemberStatus(operand.Name, operatorv1alpha1.OperatorInstalling, "", &r.Mutex)
-					continue
-				}
+				} 
 			}
 
+			csv, err := r.GetClusterServiceVersion(ctx, sub)
+
+			// If can't get CSV, requeue the request
+			if err != nil {
+				merr.Add(err)
+				requestInstance.SetMemberStatus(operand.Name, operatorv1alpha1.OperatorFailed, "", &r.Mutex)
+				continue
+			}
+
+			if csv == nil {
+				klog.Warningf("ClusterServiceVersion for the Subscription %s in the namespace %s is not ready yet, retry", operatorName, namespace)
+				requestInstance.SetMemberStatus(operand.Name, operatorv1alpha1.OperatorInstalling, "", &r.Mutex)
+				continue
+			}
+			
+			//Check csv status
+			if csv.Status.Phase != olmv1alpha1.CSVPhaseSucceeded  {
```

Comment:
Could we combine those two conditions check together by using `else if` for the second one.
```
if csv == nil {}
else if csv.Status.Phase != olmv1alpha1.CSVPhaseSucceeded {}
```